shared/actions/profile.js

```diff
@@ -462,38 +462,63 @@ function checkPgpInfoForErrors (pgpInfo: PgpInfo): PgpInfoError {
   }
 }
 
-// Returns the promise of the string that represents the public key
-function generatePgpKey (pgpInfo: PgpInfo): Promise<string> {
-  const identities = [pgpInfo.email1, pgpInfo.email2, pgpInfo.email3].map(email => ({
+// If we like this we can auto build this and the following way of calling rpc methods
+// so that they return a channel of these actions
+type IncomingKeyGenerated = NoErrorTypedAction<'keybase.1.pgpUi.keyGenerated', {key: KeyInfo}>
+type IncomingShouldPush = NoErrorTypedAction<'keybase.1.pgpUi.shouldPushPrivate', {response: {result: (shouldPush: boolean) => void}}>
+
+// Returns a channel that represents the feedback from the rpc service
+// Things in the channel look like actions
+// If the service expects a reply, a response will be attached to the payload
+function generatePgpKey (pgpInfo: PgpInfo): any {
+  const identities = [pgpInfo.email1, pgpInfo.email2, pgpInfo.email3].filter(email => !!email).map(email => ({
     username: pgpInfo.fullName || '',
     comment: '',
     email: email || '',
   }))
 
-  return new Promise((resolve, reject) => {
-    pgpPgpKeyGenRpc({
+  return eventChannel(emit => {
```

Comment:
This is pretty neat, we can convert our engine stuff into a channel that we can do all the `take` stuff so it fits well with how we use sagas.

It's also cleaner than this dispatching actions because it keeps it separate from the bigger action channel that our app uses. 

Also has the benefit of being cancellable (left as a todo until the new engine is merged). We could cancel rpc requests when we cancel a saga.